src/GoogleFit/FitOnFhir.GoogleFit/Client/Handlers/GoogleFitAuthorizationHandler.cs

```diff
@@ -84,11 +87,11 @@ private async Task<IActionResult> Authorize(RoutingRequest request)
             {
                 try
```

Comment:
Any time I see duplicate code, I think we can do something to improve reuse. If we change the order of the if statements in the `Evaluate()` method to check if the request is a Callback request first, we can do the token validation, create the state object, and pass the `state` and `token` to the `Authorize()` and `Revoke()` methods to handle them. This probably reduces the class size by 20 lines.